scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/JgitServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.io.File;
+import java.io.PrintWriter;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.JGitInternalException;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.lib.TextProgressMonitor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.scriptoocloud.core.exceptions.InvalidGitUrlException;
+import edu.asu.diging.scriptoocloud.core.service.DeleteFilesService;
+import edu.asu.diging.scriptoocloud.core.service.JgitService;
+
+@Service
+class JgitServiceImpl implements JgitService{
+
+    @Autowired
+    private DeleteFilesService deleteFilesService;
+
+    @Override
+    public void clone(String filename, String url) throws InvalidGitUrlException{
```

Comment:
what is `filename`? the name of the directory to clone to? needs a better variable name. Also, this class needs javadoc.